Fig.Tests/SqlProviderTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Fig.SqlSettings;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.Data.SQLite;
+
+namespace Fig.Test
+{
+    public class SqlProviderTests
+    {
+        private readonly Dictionary<string, string> _defaultSettingValues = new Dictionary<string, string>
+        {
+            {"DataDrive", "D:/" },
+            {"CompanyName", "DevrexLabs" }
+        };
+
+        private SqlSettingsSource _settings;
+
+        [SetUp]
+        public void SetUp()
+        {
+            using (var con = GetPrefilledMemoryDatabase())
+            {
+                _settings = new SqlSettingsSource("", "", "SELECT Key, Value FROM Settings");
+            }  
+        }
+
+        [Test]
+        public void SqlSettingsSourceGetSettingsCorrect()
+        {
+            var settings = _settings
+                .ToSettingsDictionary();
+
+            foreach (var kvp in _defaultSettingValues)
+            {
+                Assert.AreEqual(kvp.Value, settings[kvp.Key]);
+            }
+        }
+
+        private SQLiteConnection GetPrefilledMemoryDatabase()
```

Comment:
Anyway we can pre-setup memory db and use it in dbproviderfactory?

Reply:
If we go back to your original solution then problem solved. If we stick with DbProviderFactories then I suggest adding an internal constructor that takes an IDbConnection. Add an `InternalsVisibleTo()` to the project and reference the test project, now you have access to the internal constructor from the test project! (Sorry if you know all this already..)

Reply:
Passing a connection through the internal constructor would make the test not really "test" creating a connection with the dbfactory so I'm not sure if this is a good idea.

Reply:
I suppose building the insert query and appending it with the select query could be an option?